llama_stack/templates/huggingface/huggingface.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from pathlib import Path
+
+from llama_stack.distribution.datatypes import Provider
+from llama_stack.providers.inline.post_training.huggingface import HuggingFacePostTrainingConfig
+from llama_stack.providers.remote.datasetio.huggingface import HuggingfaceDatasetIOConfig
+from llama_stack.templates.template import DistributionTemplate, RunConfigSettings
+
+
+def get_distribution_template() -> DistributionTemplate:
+    providers = {
+        "post_training": ["inline::huggingface"],
+        "datasetio": ["inline::localfs", "remote::huggingface"],
+    }
+    name = "huggingface"
+    datasetio_provider = Provider(
+        provider_id="huggingface",
+        provider_type="remote::huggingface",
+        config=HuggingfaceDatasetIOConfig.sample_run_config(f"~/.llama/distributions/{name}"),
+    )
+    posttraining_provider = Provider(
+        provider_id="huggingface",
+        provider_type="inline::huggingface",
+        config=HuggingFacePostTrainingConfig.sample_run_config(f"~/.llama/distributions/{name}"),
+    )
+
+    return DistributionTemplate(
+        name="huggingface",
+        distro_type="self_hosted",
+        description="Use HuggingFace SFTTrainer for post_training",
+        container_image=None,
+        template_path=Path(__file__).parent / "doc_template.md",
+        providers=providers,
+        available_models_by_provider=None,
+        run_configs={
+            "run.yaml": RunConfigSettings(
+                provider_overrides={
+                    "post_training": [posttraining_provider],
+                    "datasetio": [datasetio_provider],
+                },
+                default_models=None,
+            ),
+        },
+        run_config_env_vars={
+            "HF_DEVICE": (
```

Comment:
We may want to have some consistency across torch inline providers. Maybe rename this into `TRAINING_DEVICE`? Then the same name can be used for torchtune and others.